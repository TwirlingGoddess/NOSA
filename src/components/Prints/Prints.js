import React from 'react';
import { NavLink } from 'react-router-dom';

export const Prints = () => (
  <div className="Prints">
    <article>
      <NavLink to="/posters">Posters</NavLink>
    </article>
    <article>
      <NavLink to="/postcards">Post Cards</NavLink>
    </article>
    <article>
      <NavLink to="/giftcards">Gift Cards</NavLink>
    </article>
    <article>
      <NavLink to="/stickers">Stickers</NavLink>
    </article>
  </div>
);
