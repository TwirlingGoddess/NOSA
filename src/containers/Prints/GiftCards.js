import React from 'react';

export const GiftCards = () => {
  return(
      <div className="GiftCards">
        <h1>pic a debunk image for your card</h1>
        <h2>enter editable text</h2>
      </div>
    )
}