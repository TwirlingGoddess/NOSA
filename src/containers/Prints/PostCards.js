import React from 'react';

export const PostCards = () => {
  return(
    <div>
      <div className="PostCards">
      </div>
      <button>Add Debunked Image</button>
      <button>Add Text</button>
    </div>
  )
}