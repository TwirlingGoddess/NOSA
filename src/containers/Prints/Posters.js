import React from 'react';

export const Posters = () => {
  return(
      <div className="Posters">
        <h1>pic a debunk image for your card</h1>
        <h2>enter editable text</h2>
      </div>
    )
}