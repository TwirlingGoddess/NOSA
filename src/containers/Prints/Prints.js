import React, { Component } from 'react';
import { connect } from 'react-redux'

class Prints extends Component {
  render() {
    return(
      <div>
        <article>Posters</article>
        <article>Post Cards</article>
        <article>Gift Cards</article>
        <article>Stickers</article>
      </div>
    )
  }
}

export default connect(null, null)(Prints)