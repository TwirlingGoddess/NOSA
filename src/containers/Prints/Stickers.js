import React, { Component } from 'react';
import { connect } from 'react-redux';

class Stickers extends Component {
  constructor() {
    super()
    this.state = {
      default: {
        title:'YOU\'RE CRAZY!!!',
        text: '...if you think that the Earth is ROUND!??!',
        id: -1
      },

      messages: [
        {
          title: 'Knock Knock... Who\'s there?',
          text: 'EARTH IS FLAT!!!',
          id: 0
        },
        {
          title: 'BABY... GOT... FLAT!',
          text: 'I like flat earth and I cannot lie',
          id: 1
        },
        {
          title: 'If I could be a frisbee...',
          text: 'I WOULD BE THE EARTH',
          id: 2
        },
        {
          title: 'When I say FLAT, you say EARTH',
          text: 'FLAT!!  EARTH!!  FLAT!!  EARTH!!',
          id: 3
        }
      ],

      currentPrintPic: '',
      hidden: 'sure'
    }
  }

  changeText = () => {
    const updatedDefault = this.state.messages.find((message, index) => message.id > this.state.default.id)
    this.setState({
      default: updatedDefault
    })
    if(this.state.default.id >= 3) {
      this.setState({
        default: {
          title:'YOU\'RE CRAZY!!!',
          text: '...if you think that the Earth is ROUND!??!',
          id: -1
        }
      })
    }
  }

  toggleDebunksDisplay = () => {
    this.setState({
      hidden: 'nope'
    })
  }

  addPicToPrint = (url) => {
    this.setState({
      currentPic: url
    })
  }

  printOut = () => {
    window.print()
  }

  render() {
    const newImages = this.props.debunks.map(image => <img onClick={() => this.addPicToPrint(image.url)} alt='' src={image.url}/>)
    return(
      <div>
        <div className="DebunksContainer">
          <div className = "TemplateContainer">
            <div className="Stickers">
              <h1>{this.state.default.title}</h1>
              <img className='currentPrintPic' alt='' src={this.state.currentPic} />
              <h3>{this.state.default.text}</h3>
            </div>
            <button onClick={this.toggleDebunksDisplay}>Show Debunked Images</button>
            <button onClick={this.changeText}>Change Text</button>
            <button onClick={this.printOut}>PRINT</button>
          </div>
          <div className={this.state.hidden}>
            {newImages}
          </div>
        </div>
      </div>
    )
  }
}

export const mapStateToProps = ({debunks}) => ({debunks})

export default connect(mapStateToProps, undefined)(Stickers)
