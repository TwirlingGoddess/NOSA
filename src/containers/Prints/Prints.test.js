import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { Prints } from './Prints';

describe('Prints', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Prints />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});