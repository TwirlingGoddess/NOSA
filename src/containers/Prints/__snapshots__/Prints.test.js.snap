// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prints should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <article>
          <NavLink
                    activeClassName="active"
                    aria-current="page"
                    to="/posters"
          >
                    Posters
          </NavLink>
</article>,
        <article>
          <NavLink
                    activeClassName="active"
                    aria-current="page"
                    to="/postcards"
          >
                    Post Cards
          </NavLink>
</article>,
        <article>
          <NavLink
                    activeClassName="active"
                    aria-current="page"
                    to="/giftcards"
          >
                    Gift Cards
          </NavLink>
</article>,
        <article>
          <NavLink
                    activeClassName="active"
                    aria-current="page"
                    to="/stickers"
          >
                    Stickers
          </NavLink>
</article>,
      ],
      "className": "Prints",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            aria-current="page"
            to="/posters"
>
            Posters
</NavLink>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "Posters",
            "to": "/posters",
          },
          "ref": null,
          "rendered": "Posters",
          "type": [Function],
        },
        "type": "article",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            aria-current="page"
            to="/postcards"
>
            Post Cards
</NavLink>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "Post Cards",
            "to": "/postcards",
          },
          "ref": null,
          "rendered": "Post Cards",
          "type": [Function],
        },
        "type": "article",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            aria-current="page"
            to="/giftcards"
>
            Gift Cards
</NavLink>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "Gift Cards",
            "to": "/giftcards",
          },
          "ref": null,
          "rendered": "Gift Cards",
          "type": [Function],
        },
        "type": "article",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            aria-current="page"
            to="/stickers"
>
            Stickers
</NavLink>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "Stickers",
            "to": "/stickers",
          },
          "ref": null,
          "rendered": "Stickers",
          "type": [Function],
        },
        "type": "article",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <article>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        to="/posters"
            >
                        Posters
            </NavLink>
</article>,
          <article>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        to="/postcards"
            >
                        Post Cards
            </NavLink>
</article>,
          <article>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        to="/giftcards"
            >
                        Gift Cards
            </NavLink>
</article>,
          <article>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        to="/stickers"
            >
                        Stickers
            </NavLink>
</article>,
        ],
        "className": "Prints",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              aria-current="page"
              to="/posters"
>
              Posters
</NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Posters",
              "to": "/posters",
            },
            "ref": null,
            "rendered": "Posters",
            "type": [Function],
          },
          "type": "article",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              aria-current="page"
              to="/postcards"
>
              Post Cards
</NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Post Cards",
              "to": "/postcards",
            },
            "ref": null,
            "rendered": "Post Cards",
            "type": [Function],
          },
          "type": "article",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              aria-current="page"
              to="/giftcards"
>
              Gift Cards
</NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Gift Cards",
              "to": "/giftcards",
            },
            "ref": null,
            "rendered": "Gift Cards",
            "type": [Function],
          },
          "type": "article",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              aria-current="page"
              to="/stickers"
>
              Stickers
</NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Stickers",
              "to": "/stickers",
            },
            "ref": null,
            "rendered": "Stickers",
            "type": [Function],
          },
          "type": "article",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
